src/components/Message/MessageSimple/MessageContent.js

```diff
@@ -93,683 +82,592 @@ const FailedText = styled.Text`
-        'Please use this.props.openReactionPicker instead.',
-    );
-
-    await this.props.openReactionPicker();
+  const onShowActionSheet = async () => {
+    await dismissKeyboard();
+    setActionSheetVisible(true);
+    if (actionSheetRef.current) {
+      actionSheetRef.current.show();
+    }
   };
 
-  onActionPress = (action) => {
+  const onActionPress = async (action) => {
     switch (action) {
       case MESSAGE_ACTIONS.edit:
-        this.handleEdit();
+        handleEdit();
         break;
       case MESSAGE_ACTIONS.delete:
-        this.handleDelete();
+        await handleDelete();
         break;
       case MESSAGE_ACTIONS.reply:
-        this.openThread();
+        onOpenThread();
         break;
       case MESSAGE_ACTIONS.reactions:
-        this.props.openReactionPicker();
+        openReactionPicker();
         break;
       default:
         break;
     }
+    setActionSheetVisible(false);
   };
 
-  render() {
-    const {
-      alignment,
-      message,
-      isMyMessage,
-      readOnly,
-      disabled,
-      Message,
-      ReactionList,
-      handleReaction,
-      hideReactionCount,
-      hideReactionOwners,
-      threadList,
-      retrySendMessage,
-      messageActions,
-      groupStyles,
-      additionalTouchableProps,
-      reactionsEnabled,
-      getTotalReactionCount,
-      repliesEnabled,
-      canEditMessage,
-      canDeleteMessage,
-      MessageHeader,
-      MessageFooter,
-      supportedReactions,
-      openReactionPicker,
-      dismissReactionPicker,
-      reactionPickerVisible,
-      handleAction,
-      AttachmentFileIcon,
-      MessageText,
-      channel,
-      MessageReplies,
-      AttachmentActions,
-      Card,
-      CardHeader,
-      CardCover,
-      CardFooter,
-      UrlPreview,
-      Giphy,
-      Gallery,
-      FileAttachment,
-      FileAttachmentGroup,
-      t,
-      tDateTimeParser,
-      markdownRules,
-    } = this.props;
-
-    const Attachment = this.props.Attachment;
-    const hasAttachment = Boolean(
-      message && message.attachments && message.attachments.length,
+  const hasAttachment = Boolean(
+    message && message.attachments && message.attachments.length,
+  );
+  const showTime = groupStyles[0] === 'single' || groupStyles[0] === 'bottom';
+  const hasReactions =
+    reactionsEnabled &&
+    message.latest_reactions &&
+    message.latest_reactions.length > 0;
+  const options = [{ id: 'cancel', title: 'Cancel' }];
+  const images =
+    hasAttachment &&
+    message.attachments.filter(
+      (item) =>
+        item.type === 'image' && !item.title_link && !item.og_scrape_url,
     );
+  const files =
+    hasAttachment && message.attachments.filter((item) => item.type === 'file');
+
+  if (
+    messageActions &&
+    reactionsEnabled &&
+    messageActions.indexOf(MESSAGE_ACTIONS.reactions) > -1
+  ) {
+    options.splice(1, 0, {
+      id: MESSAGE_ACTIONS.reactions,
+      title: t('Add Reaction'),
+    });
+  }
```

Comment:
My thought was to keep the instantiation of `options` at the MessageContent level and then pass our default values into the ActionSheet in case someone customizing the component still wants to use our option values. If we create the options object in the component then those customizing the component won't have access to our defaults. I'll remove remove `options` as an action sheet prop now, just let me know if you have second thoughts on it